app/types/types.ts

```diff
@@ -0,0 +1,99 @@
+import { Model, Document } from "mongoose";
+import { STVEvent } from '../algorithms/stv';
+
+export enum ElectionSystems {
+    NORMAL = 'normal',
+    STV = 'stv',
+}
+
+export enum Status {
+    resolved = 'RESOLVED',
+    unresolved = 'UNRESOLVED',
+}
+
+export type Count = { [key: string]: number };
+
+export type ElectionResult = {
+    result: STVResult | NormalResult;
+    thr: number;
+    seats: number;
+    voteCount: number;
+    blankVoteCount: number;
+    useStrict: boolean;
+    log: STVEvent[] | Count;
+};
+
+export interface STVResult {
+    status: Status;
+    winners: AlternativeType[];
+}
+
+export interface NormalResult extends STVResult {
+    status: Status;
+    winners: AlternativeType[];
+    count: number;
+}
+
+interface IAlternative {
+    _id: string;
+    description: string;
+    election: ElectionType;
+}
+export type AlternativeType = IAlternative & Document;
+
+interface IElection {
+    _id: string;
+    title: string;
+    description: string;
+    active: boolean;
+    hasVotedUsers: UserType[];
+    alternatives: AlternativeType[];
+    seats: number;
+    votes: VoteType[];
+    type: string;
+    useStrict: boolean;
+    accessCode: number;
+
+    // methods
+    elect() : ElectionResult;
+    addAlternative(alternative: AlternativeType) : AlternativeType;
+    addVote(user: UserType, priorities: AlternativeType[]) : AlternativeType;
+}
+export type ElectionType = IElection & Document;
+
+interface IRegister {
+    _id: string;
+    identifier: string;
+    email: string;
+    user: UserType;
+}
+export type RegisterType = IRegister & Document;
+
+interface IUser extends Document {
+    _id: string;
```

Comment:
`_id` is included in `Document`, so is there a reason it's explicitly added here?